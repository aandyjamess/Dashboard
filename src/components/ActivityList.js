import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

import * as copy from "./../copy";

const ActivityList = ({actions, groupIndex, group}) => (
  <div className="activity panel">
    <div className="activity-heading panel-heading" />
    {group.map(({transaction}, index) => (
      <div
        key={groupIndex + index}
        className="activity-item clearfix"
        onClick={actions.openModal(groupIndex + index)}
      >
        <div className="activity-check pull-left">
          <icon className="icon icon-activity_tick" />
        </div>
        <div className="activity-avatar icon pull-left icon-user_selfie_ph" />
        <div className="activity-date pull-right">
          {moment(transaction["unix-timestamp"] * 1000).format(copy.dateFormat.activity)}
        </div>
      </div>)
    )}
  </div>
);

ActivityList.PropTypes = {
  groupIndex: PropTypes.number,
  group: PropTypes.array,
};

ActivityList.defaultProps = {
  groupIndex: 0,
  group: []
};

export default ActivityList;
