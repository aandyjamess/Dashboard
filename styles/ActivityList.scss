@import "./variables";

.panel {
  border-radius: 0;
  border: 1px solid $color-15;
  margin-bottom: $gutter;
  font-size: $font-size-normal;
  font-weight: normal;
  box-shadow: none;
}

.activity {
  &-item {
    padding: 15px 15px;
    display: flex;
    align-items: center;
    cursor: pointer;

    &:hover {
      background-color: $color-15;
    }
  }

  &-heading {
    padding: 15px 15px;
    background-color: $color-16;
    color: $color-3;
  }

  &-check, &-avatar, &-text  {
    margin-right: 15px;
  }

  &-avatar {
    margin-right: 40px;
  }

  &-date {
    margin-left: auto;
  }
}
