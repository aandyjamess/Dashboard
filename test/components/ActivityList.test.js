import React from "react";
import toJson from "enzyme-to-json";

import { shallow } from "enzyme";

import ActivityList from "../../src/components/ActivityList";

describe("ActivityList", () => {
  let mockFn, wrapper;

  beforeEach(() => {
    mockFn = jest.fn();
    const actions = {openModal: () => mockFn};
    const group = [{"transaction": {"unix-timestamp": 0}}];
    wrapper = shallow(<ActivityList actions={actions} group={group} />);
  });

  it("renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders activities correctly", () => {
    expect(wrapper.find(".activity-item").length).toBe(1);
  })

  it("dispatches openModal event", () => {
    wrapper.find(".activity-item").simulate("click");

    expect(mockFn.mock.calls.length).toBe(1);
  });
});
