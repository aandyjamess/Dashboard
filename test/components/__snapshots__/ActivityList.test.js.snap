// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList renders correctly 1`] = `
<div
  className="activity panel"
>
  <div
    className="activity-heading panel-heading"
  />
  <div
    className="activity-item clearfix"
    onClick={[Function]}
  >
    <div
      className="activity-check pull-left"
    >
      <icon
        className="icon icon-activity_tick"
      />
    </div>
    <div
      className="activity-avatar icon pull-left icon-user_selfie_ph"
    />
    <div
      className="activity-date pull-right"
    >
      12:00 01 1 January 1970
    </div>
  </div>
</div>
`;
